### PR TITLE
Integrate the bootstrap.sh functionality into the Makefile.

### PR DESCRIPTION
**Pull request by petermattis**
Bootstrapping is re-run whenever the git hooks change or the
build/devbase/godeps.sh script changes.

### CODE REVIEW

File: build/devbase/godeps.sh [hunk 1]
```diff
@@ -7,11 +7,24 @@ set -e
 # by more). Adding $GOPATH/bin to $PATH is common but optional
 # so add it here to make sure we can find glock.
```
Comment (bdarnell) [inline build/devbase/godeps.sh]: Also remove the last sentence of this comment.

File: Makefile [hunk 2]
```diff
@@ -118,11 +118,26 @@ clean:
...
+	@mkdir -p $(dir $@) 
+	@ln -s ../../$(basename $<) $(dir $@) 
+
+# Update the git hooks and run the bootstrap script whenever either
+# one changes.
+.bootstrap: $(GITHOOKS) build/devbase/godeps.sh
+	@build/devbase/godeps.sh
+	@touch $@
+
+-include .bootstrap
```
Comment (bdarnell) [inline Makefile]: What does this include do?

**Comment by bdarnell** [comment]
LGTM

File: build/devbase/godeps.sh (hunk 1 above)
Comment (petermattis) [inline build/devbase/godeps.sh]: Done.

File: Makefile (hunk 2 above)
Comment (petermattis) [inline Makefile]: The `.bootstrap` file is empty, but we want make to check to make sure it is up to date every time it is run. The `-include .bootstrap` is a trick that tells make to include the named file and as a side-effect make will make sure that target is up to date.

**Pull request merged**